network/codec/cbor/encoder.go

```diff
@@ -0,0 +1,43 @@
+// (c) 2019 Dapper Labs - ALL RIGHTS RESERVED
+
+package cbor
+
+import (
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+
+	_ "github.com/onflow/flow-go/utils/binstat"
+)
+
+// Encoder is an encoder to write serialized JSON to a writer.
+type Encoder struct {
+	enc *cbor.Encoder
+}
+
+// Encode will convert the given message into binary JSON and write it to the
+// underlying encoder, followed by a new line.
+func (e *Encoder) Encode(v interface{}) error {
+
+	// encode the value
+	data, code, err := v2envEncode(v, ":strm<1(cbor)")
+	if err != nil {
+		return fmt.Errorf("could not encode value: %w", err)
+	}
+
+	if (code <= CodeMin) || (code >= CodeMax) {
+		return fmt.Errorf("invalid message envelope code: %d", code)
+	}
+
+	data = append(data, code)
```

Comment:
Just checking don't we need any delimiter to demarcate the data from code?

Reply:
@yhassanzadeh13 thanks for the comment!

FYI in theory then no delimiter is needed. Why? Because it's just an array of bytes and the code byte is always the last byte. That means upon decoding then the last byte is always the code byte and the CBOR bytes to become unmarshalled are all the other bytes. This is an optimization for CBOR and possible because everything is just an array of bytes. Whereas, the JSON code does an expensive second marshal and second unmarshal to wrap the code in an 'envelope'.

As a side note, it turns out that appending the code is not for free either, although much faster than doing the second marshal with the envelope technique. However, I have discovered a yet faster way -- documented in Notion -- to include the code and intend to push that optimization with a future PR which actually enables CBOR.